### PR TITLE
Feature/support run provider

### PR DESCRIPTION
**Pull request by bxy4543**
1.  sealer run support specific infra provider #814 
2. fix kubeadm template error caused by ip port

### CODE REVIEW

File: sealer/cmd/run.go [hunk 1]
```diff
@@ -52,6 +55,7 @@ create cluster to your baremetal server, appoint the iplist:
 func init() {
 	runArgs = &common.RunArgs{}
 	rootCmd.AddCommand(runCmd)
+	runCmd.Flags().StringVarP(&runArgs.Provider, "provider", "", "", "set infra provider, example `ALI_CLOUD`, the local server need ignore this")
```
Comment (kakaZhou719) [inline sealer/cmd/run.go:58]: short args: "-p" ?

File: apply/run.go [hunk 2]
```diff
@@ -83,35 +76,40 @@ func (c *ClusterArgs) SetClusterArgs() error {
...
 	}
-	if IsNumber(c.masterArgs) && (IsNumber(c.nodeArgs) || c.nodeArgs == "") {
-		c.cluster.Spec.Masters.Count = c.masterArgs
-		c.cluster.Spec.Nodes.Count = c.nodeArgs
-		c.cluster.Spec.Provider = common.DefaultCloudProvider
-	} else if IsIPList(c.masterArgs) && (IsIPList(c.nodeArgs) || c.nodeArgs == "") {
-		c.cluster.Spec.Masters.IPList = strings.Split(c.masterArgs, ",")
-		if c.nodeArgs != "" {
-			c.cluster.Spec.Nodes.IPList = strings.Split(c.nodeArgs, ",")
+	if IsNumber(c.runArgs.Masters) && (IsNumber(c.runArgs.Nodes) || c.runArgs.Nodes == "") {
```
Comment (kakaZhou719) [inline apply/run.go:95]: if utils.InList(c.runArgs.Provider, []string{common.AliCloud, common.CONTAINER}) {
c.cluster.Spec.Provider = c.runArgs.Provider
}
i think here we don’t  need to double check the value of c.runArgs.Provider

File: sealer/cmd/run.go (hunk 1 above)
Comment (bxy4543) [reply sealer/cmd/run.go:58]: `passwd` already use `-p`

File: apply/run.go (hunk 2 above)
Comment (bxy4543) [reply apply/run.go:95]: If the provider is a Cloud mod but the masters or Nodes type is set to ipList,  i think need to check this.

Comment (bxy4543) [reply apply/run.go:95]: If the provider is a Cloud mod but the masters or Nodes type is set to ipList,  i think need to check this.

**Comment by fanux** [comment]
Maybe we can use the cobra validate function to validate args.

**Review by fanux — APPROVED** [review]

**Pull request merged**